### ISSUE TITLE
Allow sharing datasets

### ISSUE THREAD

**Issue by jonc125** [ModellingWebLab/WebLab#206]
We have 'collaborators' functionality for entities but not datasets at present. This is probably low priority, but I can imagine people wanting to share private datasets with colleagues for viewing purposes!

Probably easier to do after #203.

**Comment by jonc125** [comment]
Duplicated by #247, but that has more detail.
